### PR TITLE
[feature](nereids) dump physical tree and memo 

### PR DESCRIPTION
**Pull request by englefly**
# Proposed changes

dump memo info and physical plan in stdout and log
set `enable_nereids_trace` variable true/false to open/close this dump.

following is a fragment of memo:
```
Group[GroupId#8]
GroupId#8(plan=PhysicalHashJoin ( type=INNER_JOIN, hashJoinCondition=[(r_regionkey#250 = n_regionkey#255)], otherJoinCondition=Optional.empty, stats=null )) children=[GroupId#6 GroupId#7 ] stats=(rows=25, isReduced=false, width=2)
GroupId#8(plan=PhysicalHashJoin ( type=INNER_JOIN, hashJoinCondition=[(r_regionkey#250 = n_regionkey#255)], otherJoinCondition=Optional.empty, stats=null )) children=[GroupId#7 GroupId#6 ] stats=(rows=25, isReduced=false, width=2)
```

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by englefly** [comment]
@morrySnow @jackwener please review

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 1]
```diff
@@ -200,6 +200,7 @@ public class SessionVariable implements Serializable, Writable {
     public static final String ENABLE_NEREIDS_RUNTIME_FILTER = "enable_nereids_runtime_filter";
 
     public static final String NEREIDS_STAR_SCHEMA_SUPPORT = "nereids_star_schema_support";
+    public static final String ENABLE_NEREIDS_DEBUG = "enable_nereids_debug";
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:203]: debug -> trace

File: fe/fe-core/src/main/java/org/apache/doris/statistics/StatsDeriveResult.java [hunk 2]
```diff
@@ -146,4 +146,13 @@ public StatsDeriveResult merge(StatsDeriveResult other) {
     public StatsDeriveResult copy() {
         return new StatsDeriveResult(this);
     }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+        builder.append(" stats=(rows:").append(rowCount)
+                .append(" isReduced:").append(isReduced)
+                .append(" width:").append(width).append(")");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/StatsDeriveResult.java:155]: i think we'd better to keep string format like other class.
```
xxx (k = v, ...)
```

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHashJoin.java [hunk 3]
```diff
@@ -88,6 +88,16 @@ public <R, C> R accept(PlanVisitor<R, C> visitor, C context) {
...
     @Override
     public String toString() {
+        if (statsDeriveResult != null) {
+            return Utils.toSqlString("PhysicalHashJoin",
+                    "type", joinType,
+                    "hashJoinCondition", hashJoinConjuncts,
+                    "otherJoinCondition", otherJoinCondition,
+                    "estimateRows", statsDeriveResult.getRowCount(),
+                    "isReduced", statsDeriveResult.isReduced,
+                    "width", statsDeriveResult.width
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHashJoin.java:98]: why print stats info in plans? i don't think this is a good idea

Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/physical/PhysicalHashJoin.java:98]: it help us to check whether the plan is as good as we expected.

**Comment by morrySnow** [comment]
please add a debug string example in commit msg~

**Review by morrySnow — APPROVED** [review]

**Pull request merged**